ui/src/app/base/types.ts

```diff
@@ -200,105 +215,131 @@ export type PodHintExtras = {
   local_disks: number;
 };
 
-// TODO: Machines and Controllers are almost identical save a few properties,
-// the Host type should represent the intersection with Machine and Controller
-// only adding the appropriate specific types e.g. controller.version
-export type Host = Machine | Controller;
-
-export type Machine = {
+/**
+ * SimpleNode represents the intersection of Devices, Machines and Controllers
+ */
+export type SimpleNode = {
   actions: string[];
+  domain: ModelRef;
+  hostname: string;
+  fqdn: string;
+  link_type: string;
+  node_type_display: string;
+  permissions: string[];
+  system_id: string;
+  tags: string[];
+} & Model;
+
+export type Device = {
+  extra_macs: string[];
+  fabrics: string[];
+  ip_address?: string;
+  ip_assignment: "external" | "dynamic" | "static";
+  link_speeds: number[];
+  owner: string;
+  parent?: string;
+  primary_mac: string;
+  spaces: string[];
+  subnets: string[];
+  zone: ModelRef;
+} & SimpleNode;
+
+type NodeStatus =
```

Comment:
I wonder if we need to define all these. It's probably fine, but as it's API data I wonder if we'd catch any errors by having these values at compile time...

Reply:
Nice for tests when we're setting `<node>.status` as we'll get valid autocompletions:

![image](https://user-images.githubusercontent.com/130286/86553308-7b254700-bf9e-11ea-8ffc-fcec1187a54d.png)

Reply:
I guess that's a case that the data doesn't really change, but we should probably be careful of not adding too much dynamic data that might give us issues and a maintenance burden down the track.